Add export section for residuals and predictions

JohnnyDoorn opened the pull request:
Fix https://github.com/jasp-stats/jasp-issues/issues/2989
Fix https://github.com/jasp-stats/jasp-issues/issues/2931

JohnnyDoorn commented:
OK just managed to test it in dev module and for me it works - could you please review @kylelang?

kylelang reviewed (commented):
The fix for https://github.com/jasp-stats/jasp-issues/issues/2931 looks good.

There are two potential issues that might need to be addressed:

1. The `export` option correctly saves *raw* residuals to the dataset, but https://github.com/jasp-stats/jasp-issues/issues/2989 requested the ability to save *standardized* residuals. So, I don't think this PR fully satisfies the feature request.
   - Could we add an option in the QML form to allow users to specify which type of residuals they want to save (i.e., raw, standardized, studentized)?
   - Such an option might also be nice to have in **jaspRegression**.
   - Maybe this isn't an urgent need, and we can merge this PR and add the switch in a future PR.
1. The test block starting on Line 169 of "test-anovarepeatedmeasuresbayesian.R" is failing with the following error message:
> "Error while loading data. Please verify your repeated measures observations."

I'm not sure if these two issues are very important (it looks like the problem with the unit tests was already present before this PR). If the fixes can wait, then I think the PR is good to merge.

JohnnyDoorn commented:
Thanks for your time @kylelang!
Let's wait for the tests, but I think i fixed the bayesian unit test.
I also added the option to choose the type of residual, but in doing so I realized that saving predictions/residuals for rm anova did not work since it will not be a single column because of the wide data format. For now I just removed the functionality and will return to it later (for the release after hopefully)
